django/db/models/expressions.py

```diff
@@ -995,14 +995,15 @@ def set_source_expressions(self, exprs):
     def get_source_expressions(self):
         return [self.expression]
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
```

Comment:
What about an `extra_placeholders` parameter (which could be a dict) instead of `kwargs`? To me, `kwargs` isn't a descriptive name and makes the code difficult to understand. Alternatively, you could rename `**kwargs` to `**extra_placeholders` and declare other parameters like `template` as explicit kwargs?